### PR TITLE
BigQuery: add maximumBillingTier option

### LINKED ISSUE #2012: "maximumBillingTier" missing in BigQuery
The Java client library is missing the `maximumBillingTier` from the BigQuery rest API. There's no way in the SDK to set it on a per-query level for a query job configuration.

See here --> http://stackoverflow.com/questions/43647678/where-is-the-option-to-set-the-bigquery-maximum-billing-tier-in-the-new-java-sdk

And here --> https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.maximumBillingTier

It's available in the Python SDK, but not in the Java one. We need to be able to set this at a query level - **we do not want to set it project wide**.

The only workaround I see to this, is rolling back to the old version of the Java libs (the old version does expose/provide it), which is obviously a massive headache for us.

### PR DESCRIPTION
**Pull request by rborer**
Allow to pass the optional maximumBillingTier parameter (fixes #2012).

### CODE REVIEW

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java [hunk 1]
```diff
@@ -504,6 +522,11 @@ public Boolean useLegacySql() {
   }
 
   /**
+   * Return the optional billing tier limit for this job.
+   */
+  public Integer maximumBillingTier() { return maximumBillingTier; }
```
Comment (shinfan) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]:
```java
public Integer maximumBillingTier() {
  return maximumBillingTier; 
}
```

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java [hunk 2]
```diff
@@ -336,6 +342,17 @@ public Builder setUseLegacySql(Boolean useLegacySql) {
       return this;
     }
 
+    /**
+     * Limit the billing tier for this job. Queries that have resource usage beyond this tier will fail
```
Comment (shinfan) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: s/Limit/Limits/

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java [hunk 3]
```diff
@@ -504,6 +522,11 @@ public Boolean useLegacySql() {
   }
 
   /**
+   * Return the optional billing tier limit for this job.
```
Comment (shinfan) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: s/Return/Returns/

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java [hunk 4]
```diff
@@ -504,6 +522,13 @@ public Boolean useLegacySql() {
   }
 
   /**
+   * Returns the optional billing tier limit for this job.
+   */
+  public Integer maximumBillingTier() {
```
Comment (garrettjonesgoogle) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: This needs to be `getMaximumBillingTier`

Comment (shinfan) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: +1 Sorry I missed it

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java [hunk 5]
```diff
@@ -534,6 +559,7 @@ ToStringHelper toStringHelper() {
         .add("writeDisposition", writeDisposition)
         .add("dryRun", dryRun)
         .add("useLegacySql", useLegacySql)
+        .add("maximumBillingTier", maximumBillingTier)
```
Comment (garrettjonesgoogle) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: That reminds me, `hashCode` needs this too

Comment (rborer) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: Right, I forgot about it.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Comment by polleyg** [comment]
Has this been merged and rolled out?

**Comment by garrettjonesgoogle** [comment]
This is merged now. It should go out in the next release in the next few business days.

**Pull request merged**